src/IROperator.h

```diff
@@ -711,6 +727,23 @@ inline Expr min(int a, const Expr &b) {
     return Internal::Min::make(Internal::make_const(b.type(), a), b);
 }
 
+/** Returns an expression representing the lesser of an expressions
+ * vector, after doing any necessary type coersion using
+ * \ref Internal::match_types. Vectorizes cleanly on most platforms
+ * (with the exception of integer types on x86 without SSE4).
+ * The expressions are folded from right ie. min(.., min(.., ..)). */
+inline Expr min(std::vector<Expr> vec) {
+    for (auto &e : vec) {
+        user_assert(e.defined()) << "min of undefined Expr\n";
+    }
+    return Internal::fold_right(vec,
+                                [](const Expr &lhs, const Expr &rhs) -> Expr {
+                                    user_assert(lhs.type() == rhs.type())
```

Comment:
Coming in way late on this, but... why does min() assert that the types are identical, rather than calling match_types (as documented)? Deliberate or oversight?